### PR TITLE
Migrating arithmetic operators and cast function

### PR DESCRIPTION
**Pull request by qingxinhome**
**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #2565

**What this PR does / why we need it:**
According to plan2 requirements, migrating arithmetic operators.the following arithmetic operators are migrated：
 plus(+),minus(-),mult(*),div(/) IntegerDiv,Mod(%),UnaryMinus

In addition, it includes the migration of two functions:
substring(), cast()

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

**Comment by qingxinhome** [comment]
With regard to the migration of cast functions, I made generic types according to the original cast template and exposed a calling interface by means of conditional judgment. I don't know whether it is appropriate. If it is not, I will modify it

File: pkg/sql/plan2/function/builtin/multi/substr.go [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+import (
+	"github.com/matrixorigin/matrixone/pkg/container/nulls"
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/vectorize/substring"
+	"github.com/matrixorigin/matrixone/pkg/vm/process"
+)
+
+func Substring(inputVecs []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
```
Comment (m-schen) [inline pkg/sql/plan2/function/builtin/multi/substr.go:25]: I think
maybe 2 overloads for Substring is better ?
(
overload 1 : substring(string, int)
overload 2: substring(string, int, int)
)
it's doesn't better

**Review by nnsgmsone — APPROVED** [review]

File: pkg/sql/plan2/function/builtin/multi/substr.go (hunk 1 above)
Comment (qingxinhome) [reply pkg/sql/plan2/function/builtin/multi/substr.go:25]: OK, I will consider this overload method based on the number of parameters。

File: pkg/sql/plan2/function/operator/cast.go [hunk 2]
```diff
@@ -0,0 +1,930 @@
...
+func isString(t types.T) bool {
+	if t == types.T_char || t == types.T_varchar {
+		return true
+	}
+	return false
+}
+
+//  isDateSeries: return true if the types.T is date related type
+func isDateSeries(t types.T) bool {
+	if t == types.T_date || t == types.T_datetime || t == types.T_datetime {
```
Comment (yingfeng) [inline pkg/sql/plan2/function/operator/cast.go:926]: static check error
`redundant or: t == types.T_datetime || t == types.T_datetime`

**Review by yingfeng — CHANGES_REQUESTED** [review]

File: pkg/sql/plan2/function/operator/cast.go (hunk 2 above)
Comment (qingxinhome) [reply pkg/sql/plan2/function/operator/cast.go]: OK, think you,I see. I have modified and submitted.

**Pull request merged**